builtin/commit.c

```diff
@@ -211,12 +211,18 @@ static int opt_parse_deserialize(const struct option *opt, const char *arg, int
 			free(deserialize_path);
 			deserialize_path = xstrdup(arg);
 		}
-		if (deserialize_path && *deserialize_path
-		    && (wt_status_deserialize_access(deserialize_path, R_OK) != 0))
-			die("cannot find serialization file '%s'",
-			    deserialize_path);
-
-		do_explicit_deserialize = 1;
+		if (!deserialize_path || !*deserialize_path)
+			do_explicit_deserialize = 1; /* read stdin */
+		else if (wt_status_deserialize_access(deserialize_path, R_OK) == 0)
+			do_explicit_deserialize = 1; /* can read from this file */
```

Comment:
the fact that these two variables are the same made me wonder if this is the right approach, but you seem to test both cases.

Reply:
Naming things is hard.

The variable `do_implicit...` handles the case where we inherited a config setting.  This was something that the gvfs installer set up and is usually outside of the user's view.

The `do_explicit...` variable handles the `--deserialize[=<path>]` command line option, which overrides the config setting.  IIRC this was added later.

The original design was that if the path from the config was not found, silently ignore it and go on and do a scan -- since that would happen any time the gvfs service was having problems, for example, and the user wouldn't know why status was broken and they were getting an obscure error message.

But when given on the command line, usual practice is to complain if an input is not found.
But that hard error threw Jameson's test/expectations.

Hence the 2 different code paths in the original code for determining whether we can read the status-cache and whether to complain about it.

Structurally, the actual code change is a lot smaller than it looks, but I refactored it a bit for (hopefully) clarity.  Perhaps I should have written it as:

    if (path && *path && !access(...)) {
        do_implicit = 0
        do_explicit = 0
        return 0
    }